agent/engine/task_manager.go

```diff
@@ -690,3 +699,42 @@ func (mtask *managedTask) waitForStopReported() bool {
 	}
 	return taskStopped
 }
+
+// sendExecutionStoppedAtTimestamp start the timer for sending execution stopped at timestamp
+func (mtask *managedTask) sendExecutionStoppedAtTimestamp() {
+	mtask.lock.Lock()
+	defer mtask.lock.Unlock()
+	seelog.Info("Essential container stopped, start sending timestamp for stopped")
+
+	executionStoppedAtElapsed := time.Now().Sub(*mtask.Task.GetExecutionStoppedAt())
+	maxDelay := executionStoppedAtSubmitDelay - executionStoppedAtElapsed
+
+	if mtask.submitExecutionStoppedTimer != nil {
+		mtask.submitExecutionStoppedTimer.Stop()
+	}
+
+	mtask.submitExecutionStoppedTimer = mtask.time().AfterFunc(maxDelay, func() {
```

Comment:
It's possible for this timer to not be active if the agent restarts while the timer has not elapsed.  What happens then?

Reply:
On agent restart, it will resend current status [here](https://github.com/aws/amazon-ecs-agent/blob/master/agent/engine/task_manager.go#L114), the timestamp will be sent at that point.